### PR TITLE
feat(api,cli): add project variable set

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-cds#19961.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19961.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19961.1 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19961.2 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19963.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✘  
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19964.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19965.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19965.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19966.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19966.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19966.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by richardlt — APPROVED** [review]

File: cli/cdsctl/experimental_project_variableset.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+func projectVariableSetShowFunc(v cli.Values) (interface{}, error) {
+	vs, err := client.ProjectVariableSetShow(context.Background(), v.GetString(_ProjectKey), v.GetString("name"))
+	return vs, err
+}
+
+var projectVariableSetListCmd = cli.Command{
+	Name:    "list",
+	Aliases: []string{"ls"},
+	Short:   "List all variabelset in the given project",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset.go:51]:
```suggestion
	Short:   "List all variable sets in the given project",
```

File: cli/cdsctl/experimental_project_variableset.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+func projectVariableSetListFunc(v cli.Values) (cli.ListResult, error) {
+	vss, err := client.ProjectVariableSetList(context.Background(), v.GetString(_ProjectKey))
+	return cli.AsListResult(vss), err
+}
+
+var projectVariableSetDeleteCmd = cli.Command{
+	Name:    "delete",
+	Aliases: []string{"rm", "remove"},
+	Short:   "Delete a variableset on a project",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset.go:65]:
```suggestion
	Short:   "Delete a variable set on a project",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+)
+
+var projectVariableSetItemCmd = cli.Command{
+	Name:    "item",
+	Aliases: []string{"vs"},
+	Short:   "Manage item on a CDS project variableset",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:16]:
```suggestion
	Short:   "Manage item on a CDS project Variable Set",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		cli.NewCommand(projectVariableSetItemUpdateCmd, projectVariableSetItemUpdateFunc, nil, withAllCommandModifiers()...),
+		cli.NewGetCommand(projectVariableSetItemShowCmd, projectVariableSetItemShowFunc, nil, withAllCommandModifiers()...),
+	})
+}
+
+var projectVariableSetItemListCmd = cli.Command{
+	Name:    "list",
+	Aliases: []string{"ls"},
+
+	Short: "List the items of the given variableset",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:33]:
```suggestion
	Short: "List the items of the given Variable Set",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+func projectVariableSetItemListFunc(v cli.Values) (cli.ListResult, error) {
+	vs, err := client.ProjectVariableSetShow(context.Background(), v.GetString(_ProjectKey), v.GetString("variableset-name"))
+	return cli.AsListResult(vs.Items), err
+}
+
+var projectVariableSetItemShowCmd = cli.Command{
+	Name:    "show",
+	Aliases: []string{"get"},
+	Short:   "Get the given variableset item",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:50]:
```suggestion
	Short:   "Get the given variable set item",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+	Ctx: []cli.Arg{
+		{Name: _ProjectKey},
+	},
+	Args: []cli.Arg{
+		{Name: "variableset-name"},
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:55]:
```suggestion
		{Name: "variable-set-name"},
```

File: cli/cdsctl/experimental_project_variableset.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/cdsclient"
+)
+
+var projectVariableSetCmd = cli.Command{
+	Name:    "variableset",
+	Aliases: []string{"vs"},
+	Short:   "Manage VariableSet on a CDS project",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset.go:17]:
```suggestion
	Short:   "Manage Variable Set on a CDS project",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+func projectVariableSetItemShowFunc(v cli.Values) (interface{}, error) {
+	item, err := client.ProjectVariableSetItemGet(context.Background(), v.GetString(_ProjectKey), v.GetString("variableset-name"), v.GetString("item-name"))
+	return item, err
+}
+
+var projectVariableSetItemDeleteCmd = cli.Command{
+	Name:    "delete",
+	Aliases: []string{"rm", "remove"},
+	Short:   "Delete an item from a variableset",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:68]:
```suggestion
	Short:   "Delete an item from a variable set",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func projectVariableSetItemDeleteFunc(v cli.Values) error {
+	return client.ProjectVariableSetItemDelete(context.Background(), v.GetString(_ProjectKey), v.GetString("variableset-name"), v.GetString("item-name"))
+}
+
+var projectVariableSetItemCreateCmd = cli.Command{
+	Name:    "add",
+	Aliases: []string{"create"},
+	Short:   "Create a new item inside a variableset ",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:85]:
```suggestion
	Short:   "Create a new item inside a variable set",
```

File: cli/cdsctl/experimental_project_variableset_item.go [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+		}
+	}
+
+	return client.ProjectVariableSetItemAdd(context.Background(), v.GetString(_ProjectKey), v.GetString("variableset-name"), &item)
+}
+
+var projectVariableSetItemUpdateCmd = cli.Command{
+	Name:    "update",
+	Aliases: []string{""},
+	Short:   "Update an item inside a variableset ",
```
Comment (yesnault) [inline cli/cdsctl/experimental_project_variableset_item.go:137]:
```suggestion
	Short:   "Update an item inside a variable set",
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19967.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19967.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Pull request merged**